installers/linux/deb/before-upgrade.sh.erb

```diff
@@ -19,8 +19,22 @@
 <%= ERB.new(File.read(File.join(install_scripts_dir, 'shared', 'partials', "_#{name}-config-migration.sh.erb")), nil, '-', "_#{SecureRandom.hex}").result(binding) %>
 
     rm -rf /var/run/<%= name %>-running-check-from-deb-package
-    if invoke-rc.d <%= name %> status > /dev/null 2>&1; then
-        mkdir -p /var/run/<%= name %>-running-check-from-deb-package
-        touch /var/run/<%= name %>-running-check-from-deb-package/running
-        invoke-rc.d <%= name %> stop || true
+
+    stopGoAgentServiceOnUpgrade() {
+      mkdir -p /var/run/<%= name %>-running-check-from-deb-package
+      touch /var/run/<%= name %>-running-check-from-deb-package/running
+      invoke-rc.d <%= name %> stop || true
+    }
+
+    # Prior to GoCD v19.6.0 Check if go-agent is running using /etc/init.d
+    if [ -f "/etc/init.d/<%= name %>" ]; then
+      # Use /etc/init.d/SCRIPT status to determine service is running or not.
+      if /etc/init.d/<%= name %> status > /dev/null 2>&1; then
+        stopGoAgentServiceOnUpgrade
```

Comment:
Should we not be using `/etc/init.d` to stop instead of `invoke-rc.d`?

Reply:
> a bit difficult to say for sure what option was used to perform an install. Perhaps checking for `/etc/init.d/SERVICE`

@ketan -- Not sure if I understand the problem with using `/usr/share/go-agent/bin/go-agent start`.